spec/support/session_registry.rb

```diff
@@ -4,23 +4,20 @@
 require 'singleton'
 
 module Mongo
-  class Client
-    alias :get_session_without_tracking :get_session
-
-    def get_session(options = {})
-      get_session_without_tracking(options).tap do |session|
-        SessionRegistry.instance.register(session)
-      end
-    end
-  end
-
   class Session
     alias :end_session_without_tracking :end_session
 
     def end_session
       SessionRegistry.instance.unregister(self)
       end_session_without_tracking
     end
+
+    alias :materialize_without_tracking :materialize
+
+    def materialize(connection)
+      materialize_without_tracking(connection)
+      SessionRegistry.instance.register(self)
+    end
```

Comment:
Does this retain tracking for explicit sessions? It's been a while since I wrote this code, but if it previously tracked leaked explicit sessions, it should still do so going forward.